csharp/GrpcOpenTracing/GrpcOpenTracing/Handlers/InterceptedServerHandler.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Grpc.Core;
+using GrpcOpenTracing.Propagation;
+using GrpcOpenTracing.Streaming;
+using OpenTracing;
+using OpenTracing.Propagation;
+using OpenTracing.Tag;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace GrpcOpenTracing.Handlers
+{
+    internal class InterceptedServerHandler<TRequest, TResponse>
+        where TRequest : class
+        where TResponse : class
+    {
+        private readonly GrpcTraceLogger<TRequest, TResponse> logger;
+        private readonly ServerCallContext context;
+
+        public InterceptedServerHandler(ITracer tracer, ServerCallContext context)
+        {
+            this.context = context;
+
+            ISpan span = GetSpanFromContext(tracer);
+            this.logger = new GrpcTraceLogger<TRequest, TResponse>(span);
+        }
+
+        private ISpan GetSpanFromContext(ITracer tracer)
+        {
+            return GetSpanFromHeaders(tracer, context.RequestHeaders, $"Server {this.context.Method}")
+                .SetTag(Tags.Component, Constants.TAGS_COMPONENT)
+                .SetTag(Tags.SpanKind, Tags.SpanKindServer)
+                .SetTag("peer.address", this.context.Peer)
+                .SetTag("grpc.method_name", this.context.Method)
+                .SetTag("grpc.headers", GetGrpcHeaders());
+        }
+
+        private string GetGrpcHeaders()
+        {
+            return string.Join("; ", this.context.RequestHeaders.Where(e => !e.Key.Equals("x-letstrace-trace-context")).Select(e => $"{e.Key} = {e.Value}"));
+        }
+
+        private ISpan GetSpanFromHeaders(ITracer tracer, Metadata metadata, string operationName)
+        {
+            ISpan span;
+            try
+            {
+                ISpanContext parentSpanCtx = tracer.Extract(BuiltinFormats.HttpHeaders, new MetadataCarrier(metadata));
+                var spanBuilder = tracer.BuildSpan(operationName);
+                if (parentSpanCtx != null)
+                {
+                    spanBuilder = spanBuilder.AsChildOf(parentSpanCtx);
+                }
+                span = spanBuilder.StartActive(false).Span;
+            }
+            catch (Exception ex)
+            {
+                span = tracer.BuildSpan(operationName)
```

Comment:
Unnecessary try..catch IMO. The only thing that could/should fail here is if the tracer doesn't support the format (but every tracer will support `BuiltinFormats.HttpHeaders`). `Extract` should return null if the headers are malformed etc.

If you keep the try..catch then you should check the scope behavior. the try-block calls `StartActive` but the catch-block does not.